### PR TITLE
feat(synthetics): Adding DeleteLambdaResourcesOnCanaryDeletion prop to the canary L2

### PR DESCRIPTION
**Pull request by SankyRed**
This PR is to add the DeleteLambdaResourcesOnCanaryDeletion prop to the canary L2.

### CODE REVIEW

File: packages/@aws-cdk/aws-synthetics/lib/runtime.ts [hunk 1]
```diff
@@ -177,10 +187,52 @@ export class Runtime {
...
+   * `syn-python-selenium-1.1` includes the following:
+   * - Lambda runtime Python 3.8
+   * - Selenium version 3.141.0
+   * - Chromium version 83.0.4103.0
+   *
+   * New Features:
+   * - **Custom handler function**: You can now use a custom handler function for your canary scripts.
+   * - **Configuration options for adding metrics and step failure configurations**: These options were already available in runtimes for Node.js canaries.
+   * - **Custom arguments in Chrome**: You can now open a browser in incognito mode or pass in proxy server configuration.
+   * - **Cross-Region artifact buckets**: A canary can store its artifacts in an Amazon S3 bucket in a different Region.
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-synthetics/lib/runtime.ts:204]: I don't think we need this here

File: packages/@aws-cdk/aws-synthetics/lib/runtime.ts [hunk 2]
```diff
@@ -177,10 +187,52 @@ export class Runtime {
...
+
+  /**
+   * `syn-python-selenium-1.2` includes the following:
+   * - Lambda runtime Python 3.8
+   * - Selenium version 3.141.0
+   * - Chromium version 92.0.4512.0
+   *
+   * New Features:
+   * - **Updated dependencies**: The only new features in this runtime are the updated dependencies.
+   *
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-synthetics/lib/runtime.ts:218]: I also don't think we need this here

File: packages/@aws-cdk/aws-synthetics/test/canary.test.ts [hunk 3]
```diff
@@ -34,6 +34,26 @@ test('Basic canary properties work', () => {
   });
 });
 
+test('Canary deletion leads to lambda resource deletion property is set', () => {
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-synthetics/test/canary.test.ts:37]:
```suggestion
test('can set `DeleteLambdaResourceOnCanaryDeletion`', () => {
```

File: packages/@aws-cdk/aws-synthetics/lib/canary.ts [hunk 4]
```diff
@@ -204,6 +204,15 @@ export interface CanaryProps {
    * not specified a dedicated security group will be created for this canary.
    */
   readonly securityGroups?: ec2.ISecurityGroup[];
+
+  /**
+   * Whether or not the lambda resources are to be deleted when the canary is deleted
```
Comment (comcalvi) [inline packages/@aws-cdk/aws-synthetics/lib/canary.ts:209]:
```suggestion
   * Whether or not to delete the lambda resources when the canary is deleted
```

**Review by comcalvi — CHANGES_REQUESTED** [review]
Nice work! This is pretty close to being merged

File: packages/@aws-cdk/aws-synthetics/lib/runtime.ts (hunk 1 above)
Comment (SankyRed) [reply packages/@aws-cdk/aws-synthetics/lib/runtime.ts:204]: I can get these removed but, I saw the new features being mentioned within the comments for few runtime versions so added them to make it consistent.

File: packages/@aws-cdk/aws-synthetics/test/canary.test.ts (hunk 3 above)
Comment (SankyRed) [reply packages/@aws-cdk/aws-synthetics/test/canary.test.ts:37]: Will get that updated!

File: packages/@aws-cdk/aws-synthetics/lib/runtime.ts (hunk 1 above)
Comment (comcalvi) [reply packages/@aws-cdk/aws-synthetics/lib/runtime.ts:204]: ah, ok. Leave them then, we should be consistent.

File: packages/@aws-cdk/aws-synthetics/lib/canary.ts [hunk 5]
```diff
@@ -204,6 +204,15 @@ export interface CanaryProps {
...
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-synthetics-canary.html#cfn-synthetics-canary-deletelambdaresourcesoncanarydeletion
+   *
+   * @default false
+   */
+  readonly deleteLambdaResourcesOnCanaryDeletion?: boolean;
```
[the remark this answers is not in the document]
Comment (SankyRed) [reply packages/@aws-cdk/aws-synthetics/lib/canary.ts:215]: We can do that, just thought we wanted to name it as in the props. Will update it.

**Review by comcalvi — APPROVED** [review]

**Pull request merged**